### ISSUE TITLE
Histogram breaks on some values

### ISSUE THREAD

**Issue by Munksgaard** [highcharts/highcharts#7438]
#### Expected behaviour

[This jsfiddle](http://jsfiddle.net/7jntfmjL/) should work.

#### Actual behaviour

[The above jsfiddle](http://jsfiddle.net/7jntfmjL/) shows the histogram bars as very thin lines. Removing any single value from the list creates a proper histogram. [Example](http://jsfiddle.net/7jntfmjL/1/)

#### Live demo with steps to reproduce

As linked above.

#### Affected browser(s)

Chrome and Firefox

**Comment by pawelfus** [comment]
Hi @Munksgaard 

Thank you for reporting this bug. 

*Internal note:*
It looks like `derivedData` returns something wrong. We can observe in the data NaN: 

```
> Highcharts.charts[0].series[0].processedYData
> [1, 0, 0, NaN, 0]
```

Changing `0` to any other number also resolves the issue: http://jsfiddle.net/7jntfmjL/3/

**Comment by Munksgaard** [comment]
> Changing 0 in the dataset to any other number also resolves the issue: http://jsfiddle.net/7jntfmjL/3/

Not _any_ other number. For instance, changing the `0` to `0.02` still produces an error: http://jsfiddle.net/7jntfmjL/4/

**Comment by TorsteinHonsi** [comment]
Fixed by 4baa1c3ba3248c509fd569eb9411d47148ccfbc0
